easybuild/easyconfigs/t/Theano/Theano-1.0.0-intel-2017b-Python-2.7.14.eb

```diff
@@ -16,6 +16,14 @@ checksums = ['7bc82e4fcaac023cac23c390b30d5df81bc09d33568812df75b0bd2b86d2fa8f']
 
 dependencies = [('Python', '2.7.14')]
 
+# Include architecture name in compiledir to fix problems with architecture-specific compilation
+# when running the software on a heterogeneous compute cluster.
+# Change the environment variable "VSC_ARCH_LOCAL" to match your cluster setup.
+# import os
+# architecture = os.environ['VSC_ARCH_LOCAL']
+# modextravars = {'THEANO_FLAGS': 'compiledir_format=compiledir_%s-%%(short_platform)s-%%(processor)s-'
+#                  '%%(python_version)s-%%(python_bitwidth)s' % architecture}
```

Comment:
@smoors Please make the PR subject reflect what you're changing here?

I'm not sure if we should drag this along in the Theano easyconfig(s) though...

Also, the `architecture` value isn't used at all?

Reply:
I leave it up to you to decide that, but I suspect many HPC clusters may run into this issue. At least we did at VUB. Also for VSC this is highly relevant: for example, when a user from UGent tries to use Theano at VUB or vice versa.
Feel free to change the architecture name to something else.

Reply:
What kind of problems pop up without this?

Reply:
When you use the module for the first time on a node (import theano in python), an architecture-specific shared object is created at runtime in ~/.teano/compiledir_xxx. When you later try to use it on a node with an incompatible architecture, it will try to use the existing .so and fail.

Reply:
I'd say it makes more sense to somehow avoid that Theano uses `$HOME/.theano`, and make it use something like `/tmp/theano-$USER/`?

That's still sort of site-specific, but way less so than your approach...

Reply:
That should work too! Would you like to have this the default behavior, or commented out?

Reply:
I guess it makes sense to make that the default, since it would work in most cases imho.

For sites where this doesn't work, they can customise it?

It's certainly a more sensible default than using `$HOME` in an HPC context imho...

Reply:
Yes, users can always change the env var after loading, or change the default in a .theanorc file.